source/base/timer.cc

```diff
@@ -56,8 +57,10 @@ Timer::Timer()
 #endif
 {
 #ifdef DEAL_II_WITH_MPI
-  mpi_data.sum = mpi_data.min = mpi_data.max = mpi_data.avg = numbers::signaling_nan<double>();
-  mpi_data.min_index = mpi_data.max_index = numbers::invalid_unsigned_int;
+  mpi_data.sum = mpi_data.min = mpi_data.max = mpi_data.avg = 0.;
```

Comment:
At least in the signaling NaN case, the reason is always the same: if you *use* a signaling NaN, you will get an exception. That makes sense, because we initialize these variables with an *invalid* value and the first thing any code should do is overwrite this value with something valid. Using an invalid value can not ever be correct.